From 335a43c39dd22f8b261253fb0e7a1a0fd0598e29 Mon Sep 17 00:00:00 2001
From: Chih-Wei Huang <cwhuang@linux.org.tw>
Date: Sun, 30 Oct 2016 22:18:55 +0800
Subject: [PATCH 2/3] Fix a null pointer exception in server_server

It's unclear why it only happens on Android 7.0 with kernel 4.8+.
---
 services/core/java/com/android/server/AlarmManagerService.java | 1 +
 1 file changed, 1 insertion(+)

diff --git a/services/core/java/com/android/server/AlarmManagerService.java b/services/core/java/com/android/server/AlarmManagerService.java
index 20cca16..3b376a6 100644
--- a/services/core/java/com/android/server/AlarmManagerService.java
+++ b/services/core/java/com/android/server/AlarmManagerService.java
@@ -2621,6 +2621,7 @@ class AlarmManagerService extends SystemService {
                     // now trigger the alarms without the lock held
                     for (int i=0; i<triggerList.size(); i++) {
                         Alarm alarm = triggerList.get(i);
+                        if (alarm.operation == null) continue;
                         try {
                             alarm.operation.send();
                         } catch (PendingIntent.CanceledException e) {
--
2.7.4
